code/modules/shuttle/supply.dm

```diff
@@ -223,8 +226,44 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
 		msg += export_text + "\n"
 		D.adjust_money(ex.total_value[E])
 
+	SSeconomy.export_total += (D.account_balance - presale_points)
 	SSshuttle.centcom_message = msg
 	investigate_log("Shuttle contents sold for [D.account_balance - presale_points] credits. Contents: [ex.exported_atoms ? ex.exported_atoms.Join(",") + "." : "none."] Message: [SSshuttle.centcom_message || "none."]", INVESTIGATE_CARGO)
 
+/// Generates a box of mail depending on our exports and imports.
+/obj/docking_port/mobile/supply/proc/mail()
+	if(SSeconomy.mail_waiting <= 0)
```

Comment:
This proc returns bools when it shouldnt
It also unnececarily returns the crate